service/src/main/resources/HTMLTemplate.html

```diff
@@ -1066,44 +1072,52 @@ <h3 class="heading heading--two">Respondent
-          {% endif %}
-          {% if case_data.respondents.firstRespondent.address.Country is not empty %}
-            <span class="u-comma-before">{{ case_data.respondents.firstRespondent.address.Country }}</span>
+        {% if case_data.respondents is not empty and case_data.respondents.firstRespondent is not empty %}
+          {% if case_data.respondents.firstRespondent.address is not empty %}
+            {% if case_data.respondents.contactDetailsHidden is not empty and case_data.respondents.contactDetailsHidden == "Yes" %}
+              Confidential
+            {% else %}
+              {% if case_data.respondents.firstRespondent.address.AddressLine1 is not empty %}
+                <span class="u-comma-before">{{ case_data.respondents.firstRespondent.address.AddressLine1 }}</span>
+              {% endif %}
+              {% if case_data.respondents.firstRespondent.address.AddressLine2 is not empty %}
+                <span class="u-comma-before">{{ case_data.respondents.firstRespondent.address.AddressLine2 }}</span>
+              {% endif %}
+              {% if case_data.respondents.firstRespondent.address.AddressLine3 is not empty %}
+                <span class="u-comma-before">{{ case_data.respondents.firstRespondent.address.AddressLine3 }}</span>
+              {% endif %}
+              {% if case_data.respondents.firstRespondent.address.PostTown is not empty %}
+                <span class="u-comma-before">{{ case_data.respondents.firstRespondent.address.PostTown }}</span>
+              {% endif %}
+              {% if case_data.respondents.firstRespondent.address.PostCode is not empty %}
+                <span class="u-comma-before">{{ case_data.respondents.firstRespondent.address.PostCode }}</span>
+              {% endif %}
+              {% if case_data.respondents.firstRespondent.address.County is not empty %}
+                <span class="u-comma-before">{{ case_data.respondents.firstRespondent.address.County }}</span>
+              {% endif %}
+              {% if case_data.respondents.firstRespondent.address.Country is not empty %}
+                <span class="u-comma-before">{{ case_data.respondents.firstRespondent.address.Country }}</span>
+              {% endif %}
+            {% endif %}
+          {% else %}
+            -
           {% endif %}
-        {% else %}
-          -
         {% endif %}
       </dd>
     </div>
     <div class="data-list__row">
       <dt class="data-list__title">Telephone number</dt>
       <dd class="data-list__definition">
-        {% if case_data.respondents is not empty and
-        case_data.respondents.firstRespondent is not empty and
-        case_data.respondents.firstRespondent.telephone is not empty %}
-          {{ case_data.respondents.firstRespondent.telephone }}
-        {% else %}
-          -
+        {% if case_data.respondents is not empty and case_data.respondents.firstRespondent is not empty %}
```

Comment:
This should be changed as well to mark confidential only when telephone number is provided. Currently it prints Confidential label when address is not provided.

Reply:
good spot, forgot to propagate the change I did to address onto telephone